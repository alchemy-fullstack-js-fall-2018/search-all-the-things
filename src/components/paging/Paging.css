.paging {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  height: 2em;
  width: 100vw;

  button {
    width: 100px;
    background-color: blue;
    color: white;
  }
}
