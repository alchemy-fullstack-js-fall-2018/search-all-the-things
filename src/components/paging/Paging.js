import React from 'react';
import styles from './Paging.css';

export default function Paging({ currentPage, totalPages, updatePage }) {
  const previousButton = currentPage > 1 && (
    <button onClick={() => updatePage(currentPage - 1)}>&lt;</button>
  );
  const nextButton = currentPage < totalPages && (
    <button onClick={() => updatePage(currentPage + 1)}>&gt;</button>
  );

  return (
    <div className={styles.paging}>
      {previousButton}
      <span>
        {currentPage} out of {totalPages}
      </span>
      {nextButton}
    </div>
  );
}
