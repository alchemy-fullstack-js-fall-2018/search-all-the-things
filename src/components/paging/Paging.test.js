import React from 'react';
import Paging from './Paging';
import render from 'react-test-renderer';

describe('Paging', () => {
  test('paging snapshot', () => {
    expect(
      render.create(<Paging />).toJSON()

    ).toMatchSnapshot();
  });
});
