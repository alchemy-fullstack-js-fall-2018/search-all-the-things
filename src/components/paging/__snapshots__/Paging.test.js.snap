// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging paging snapshot 1`] = `
<div
  className="paging"
>
  <span>
     out of 
  </span>
</div>
`;
